library/Ivoz/Cgr/Infrastructure/Cgrates/Service/SetMaxUsageThresholdService.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Ivoz\Cgr\Infrastructure\Cgrates\Service;
+
+use Graze\GuzzleHttp\JsonRpc\ClientInterface;
+use Ivoz\Core\Infrastructure\Domain\Service\Cgrates\AbstractApiBasedService;
+
+class SetMaxUsageThresholdService extends AbstractApiBasedService
+{
+    private $companyBalanceService;
+    private $reassembleTriggerService;
+    private $enableAccountService;
+
+    public function __construct(
+        ClientInterface $jsonRpcClient,
+        CompanyBalanceService $companyBalanceService,
+        ReassembleTriggerService $reassembleTriggerService,
+        EnableAccountService $enableAccountService
+    ) {
+        parent::__construct(
+            $jsonRpcClient
+        );
+
+        $this->companyBalanceService = $companyBalanceService;
+        $this->reassembleTriggerService = $reassembleTriggerService;
+        $this->enableAccountService = $enableAccountService;
+    }
+
+    /**
+     * @return void
+     */
+    public function execute(string $tenant, string $account, int $threshold)
+    {
+        $mustReassemble = $this->isReassembleNeeded(
+            $tenant,
+            $account,
+            $threshold
+        );
+
+        $this->setNewThreshold(
+            $tenant,
+            $account,
+            $threshold
+        );
+
+        if ($mustReassemble) {
+            $this->reassembleTriggerService->execute(
+                $tenant,
+                $account
+            );
+
+            $this->enableAccountService->execute(
+                $tenant,
+                $account
+            );
+        }
+    }
+
+    private function isReassembleNeeded(string $tenant, string $account, $threshold)
+    {
+        $brandId = substr($tenant, 1);
+        $companyId = substr($account, 1);
```

Comment:
This is totally tied to how we format CGRateS ids right now. Not a problem for now.